packages/react-router5/modules/RouteProvider.js

```diff
@@ -16,24 +16,33 @@ class RouteProvider extends React.PureComponent {
         super(props)
         const { router } = props
 
+        state = {
+            mounted: false
+        }
         this.router = router
```

Comment:
The would be good to not copy this.props.router into this.router. The router should not change so we can call subscribe and unsubscribe on the same instance. We also can throw an error if the router change in componentDidUpdate()